pkg/asset/manifests/gcp/cluster.go

```diff
@@ -1 +1,150 @@
 package gcp
+
+import (
+	"fmt"
+	"net"
+	"os"
+
+	"github.com/apparentlymart/go-cidr/cidr"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+	capg "sigs.k8s.io/cluster-api-provider-gcp/api/v1beta1"
+
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/asset/installconfig"
+	"github.com/openshift/installer/pkg/asset/manifests/capiutils"
+)
+
+const (
+	// autoCreateSubnets is not used as the subnet information is provided through the install config.
+	autoCreateSubnets = false
+
+	defaultSubnetCreationPurpose = "PRIVATE_RFC_1918"
+)
+
+var (
+	subnetName = func(infraID, machineType string) string { return fmt.Sprintf("%s-%s-subnet", infraID, machineType) }
+)
+
+// GenerateClusterAssets generates the manifests for the cluster-api.
+func GenerateClusterAssets(installConfig *installconfig.InstallConfig, clusterID *installconfig.ClusterID) (*capiutils.GenerateClusterAssetsOutput, error) {
+	manifests := []*asset.RuntimeFile{}
+
+	networkName := installConfig.Config.GCP.Network
+	if installConfig.Config.GCP.NetworkProjectID != "" {
+		// Subnets should exist in the network with name NetworkProjectID when provided.
+		networkName = installConfig.Config.GCP.NetworkProjectID
+	}
+	if networkName == "" {
+		networkName = fmt.Sprintf("%s-network", clusterID.InfraID)
+	}
+
+	enableFlowLogs := os.Getenv("GCP_ENABLE_FLOW_LOGS") == "true"
+
+	masterSubnet := installConfig.Config.GCP.ControlPlaneSubnet
+	masterSubnetDescription := "Control Plane Subnet used by the installer"
+	if masterSubnet == "" {
+		masterSubnet = subnetName(clusterID.InfraID, "master")
+		masterSubnetDescription = "Control Plane Subnet owned by the installer"
+	}
+	master := capg.SubnetSpec{
+		Name:           masterSubnet,
+		CidrBlock:      "",
+		Description:    ptr.To(masterSubnetDescription),
+		Region:         installConfig.Config.GCP.Region,
+		EnableFlowLogs: ptr.To(enableFlowLogs),
+		Purpose:        ptr.To(defaultSubnetCreationPurpose),
+	}
+
+	workerSubnet := installConfig.Config.GCP.ComputeSubnet
+	workerSubnetDescription := "Compute Subnet used by the installer"
+	if workerSubnet == "" {
+		workerSubnet = subnetName(clusterID.InfraID, "worker")
+		workerSubnetDescription = "Compute Subnet owned by the installer"
+	}
+	worker := capg.SubnetSpec{
+		Name:           workerSubnet,
+		CidrBlock:      "",
+		Description:    ptr.To(workerSubnetDescription),
+		Region:         installConfig.Config.GCP.Region,
+		EnableFlowLogs: ptr.To(enableFlowLogs),
+		Purpose:        ptr.To(defaultSubnetCreationPurpose),
+	}
+
+	// Add the CIDR information.
```

Comment:
do we need to provide our own cidr info? what happens if we don't? does capg supply them?

Reply:
It does not look like CAPG does anything more than pass along the data to the compute sdk.